pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -166,11 +166,13 @@ type KVConfig struct {
 
 // RegisterFlags registers flags.
 func (cfg *KVConfig) RegisterFlags(f *flag.FlagSet, prefix string) {
+	mlDefaults := defaultMemberlistConfig()
```

Comment:
If `memberlist.DefaultLANConfig()` ever changes (eg. when we update the library), we need to be careful about handling the change in Cortex default values. I don't think it's a blocker, documenting such changes in the future should be fine.

Reply:
If the default values change from updating memberlist then `make check-doc` will fail, and we can act accordingly (e.g. add a Changelog entry, or override the values).